Publish improvements

erin-doyle opened the pull request:
This PR is to hopefully fix issue #1 .  It looks as if the `/lib` directory is not being included in the npm package.  My theory is that the use of the `release` script from rf-release is not functioning properly.  It does not seem as if the `build` script is being run before `npm publish` is called and if `prepublish` were being called then it should be.  It also doesn't look like the tags in npm are lining up with the ones in github.  It just seems rather problematic.

Additionally, it looked as if there was an intention to start using the webpack config and including a bundled js file in a `/dist` directory but that also is not getting published in the package.  I've added that script to be run during the `prepublish` script.

If `npm publish` is run manually from now on instead of using the `release` script then the `prepublish` script _should_ get run and the `build` and `bundle` scripts will be run and `/lib` and `/dist` directories should get included in the package.

Commit:
Added/removed dirs and files from npmignore

Commit:
Added the bundle script to the prepublish script so that the webpack bundle gets included in the npm package.

Also disabled the release script and removed the rf-release package as this does not seem to be tagging or publishing to npm properly.

jkomusin commented:
While we're in package.json, could we also change the `repository` entry to point to this GitHub repo? Currently the npm page is a bit confusing because of that.

Commit:
Updated urls to point to this repo instead of the forked from repo

romeovs commented:
Love this!

Could you update the README to include publish instructions?

Commit:
Added a CONTRIBUTING page and linked to it from the README.

Commit:
Added some more scripts for helping with the version and publish process and added descriptions of the scripts in the CONTRIBUTING.md

erin-doyle commented:
Np!  I've added a few more scripts that should hopefully make the publishing process _very_ smooth and I've added a CONTRIBUTING.md detailing all of it.  I have to put out the disclaimer that I wasn't able to actually test out the scripts publishing but they should work in theory.  There may need to be tweaks made to the scripts and/or the doc if anything doesn't work exactly as expected when you go to publish.

Let me know if you have any questions about the documentation, I hope it's clear and helpful!

erin-doyle commented:
Now that the publishing improvements have been merged can you try publishing a patch update so we can see if #1  is finally fixed?
